File: ios/Samples/Showcase-ObjectiveC/Showcase-ObjectiveC/ViewControllers/DDL/DDLListScreenletViewController.m
/**
 * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
 *
 * This library is free software; you can redistribute it and/or modify it under
 * the terms of the GNU Lesser General Public License as published by the Free
 * Software Foundation; either version 2.1 of the License, or (at your option)
 * any later version.
 *
 * This library is distributed in the hope that it will be useful, but WITHOUT
 * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
 * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
 * details.
 */

@import LiferayScreens;
#import "NSString+Utils.h"
#import "LiferayLogger.h"
#import "DDLListScreenletViewController.h"

@interface DDLListScreenletViewController () <DDLListScreenletDelegate,
UIPickerViewDelegate, UIPickerViewDataSource>

@property (weak, nonatomic) IBOutlet UIPickerView *pickerView;
@property (weak, nonatomic) IBOutlet DDLListScreenlet *screenlet;
@property (weak, nonatomic) IBOutlet UITextField *recordSetIdTextField;
@property (weak, nonatomic) IBOutlet UITextField *labelFieldsTextField;
@property (weak, nonatomic) IBOutlet UIButton *loadButton;

@property (nonatomic, copy) NSArray *pickerDataNames;
@property (nonatomic, copy) NSArray *pickerDataClassNames;

@property (nonatomic) NSString *selectedObcClassName;

@end

@implementation DDLListScreenletViewController

- (void)viewDidLoad {
	[super viewDidLoad];

	self.pickerDataNames = @[
		@"No order",
		@"Id",
		@"Create date",
		@"Modified Date"
	];

	self.pickerDataClassNames = @[
		@"",
		@"com.liferay.dynamic.data.lists.util.comparator.DDLRecordIdComparator",
		@"com.liferay.dynamic.data.lists.util.comparator.DDLRecordCreateDateComparator",
		@"com.liferay.dynamic.data.lists.util.comparator.DDLRecordModifiedDateComparator"
	];

	self.selectedObcClassName = @"";

	self.screenlet.delegate = self;
	self.pickerView.delegate = self;
	self.pickerView.dataSource = self;

	self.labelFieldsTextField.text = [LiferayServerContext stringPropertyForKey:@"ddlLabelField"];
	self.recordSetIdTextField.text = [NSString
									  stringWithFormat:@"%lld", [LiferayServerContext longPropertyForKey:@"ddlRecordSetId"]];
}

- (IBAction)loadList:(id)sender {
	if ([self.recordSetIdTextField.text isNumeric]) {
		self.screenlet.recordSetId = self.recordSetIdTextField.text.longLongValue;
		self.screenlet.labelFields = self.labelFieldsTextField.text;
		self.screenlet.obcClassName = self.selectedObcClassName;
		[self.screenlet loadList];
	}
}

#pragma mark DDListScreenletDelegate

- (void)screenlet:(DDLListScreenlet *)screenlet onDDLListError:(NSError *)error {
	LiferayLog(error);
}

- (void)screenlet:(DDLListScreenlet *)screenlet onDDLSelectedRecord:(DDLRecord *)record {
	LiferayLog(record);
}

- (void)screenlet:(DDLListScreenlet *)screenlet
		onDDLListResponseRecords:(NSArray<DDLRecord *> *)records {

	LiferayLog(records);
}

- (NSInteger)numberOfComponentsInPickerView:(UIPickerView *)pickerView {
	return 1;
}

- (NSInteger)pickerView:(UIPickerView *)pickerView numberOfRowsInComponent:(NSInteger)component {
	return self.pickerDataNames.count;
}

- (void)pickerView:(UIPickerView *)pickerView didSelectRow:(NSInteger)row
	   inComponent:(NSInteger)component {

	self.selectedObcClassName = self.pickerDataClassNames[row];
}

- (NSAttributedString *)pickerView:(UIPickerView *)pickerView
			 attributedTitleForRow:(NSInteger)row forComponent:(NSInteger)component {

	NSDictionary *attrs = @{NSFontAttributeName : [UIFont systemFontOfSize:3]};

	return [[NSAttributedString alloc] initWithString:self.pickerDataNames[row] attributes:attrs];
}

@end

File: ios/Samples/Showcase-swift/LiferayScreens-Showcase-Swift/ViewControllers/DDLListScreenletViewController.swift
/**
 * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
 *
 * This library is free software; you can redistribute it and/or modify it under
 * the terms of the GNU Lesser General Public License as published by the Free
 * Software Foundation; either version 2.1 of the License, or (at your option)
 * any later version.
 *
 * This library is distributed in the hope that it will be useful, but WITHOUT
 * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
 * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
 * details.
 */
import UIKit
import LiferayScreens


class DDLListScreenletViewController: UIViewController, DDLListScreenletDelegate, UIPickerViewDataSource, UIPickerViewDelegate {

	@IBOutlet weak var pickerView: UIPickerView!

	@IBOutlet weak var screenlet: DDLListScreenlet?
	@IBOutlet weak var recordSetIdTextField: UITextField?
	@IBOutlet weak var labelFieldsTextField: UITextField?

	let pickerData : [(name: String, className: String)] = [
		(name: "No order", className:""),
		(name: "Id", className: "com.liferay.dynamic.data.lists.util.comparator.DDLRecordIdComparator"),
		(name: "CreationDate", className: "com.liferay.dynamic.data.lists.util.comparator.DDLRecordCreateDateComparator"),
		(name: "ModifiedDate", className: "com.liferay.dynamic.data.lists.util.comparator.DDLRecordModifiedDateComparator")
	]

	var selectedObcClassName = ""

	@IBAction func loadList(sender: AnyObject) {
		if let recordSetId = Int(recordSetIdTextField!.text!) {
			screenlet!.recordSetId = Int64(recordSetId)
			screenlet!.labelFields = labelFieldsTextField!.text
			screenlet?.obcClassName = selectedObcClassName
			screenlet!.loadList()
		}
	}

	override func viewDidLoad() {
		super.viewDidLoad()

		self.screenlet?.delegate = self
		self.pickerView.dataSource = self
		self.pickerView.delegate = self
	}

	func screenlet(screenlet: DDLListScreenlet,
			onDDLListResponseRecords records: [DDLRecord]) {
		print("DELEGATE: onDDLListResponse called -> \(records)\n");
	}

	func screenlet(screenlet: DDLListScreenlet,
			onDDLListError error: NSError) {
		print("DELEGATE: onDDLListError called -> \(error)\n");
	}

	func screenlet(screenlet: DDLListScreenlet,
			onDDLSelectedRecord record: DDLRecord) {
		print("DELEGATE: onDDLRecordSelected called -> \(record)\n");
	}

	// MARK: UIPickerViewDataSource

	func numberOfComponentsInPickerView(pickerView: UIPickerView) -> Int {
		return 1
	}

	func pickerView(pickerView: UIPickerView, numberOfRowsInComponent component: Int) -> Int {
		return pickerData.count
	}

	func pickerView(pickerView: UIPickerView, didSelectRow row: Int, inComponent component: Int) {
		selectedObcClassName = pickerData[row].className
	}

	func pickerView(pickerView: UIPickerView, attributedTitleForRow row: Int, forComponent component: Int) -> NSAttributedString? {
		let attrs = [NSFontAttributeName : UIFont.systemFontOfSize(3)]

		return NSAttributedString(string: pickerData[row].name, attributes: attrs)
	}
}
